test/utils.jl

```diff
@@ -1,6 +1,65 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 import Pkg
 
+const LOADED_DEPOT = joinpath(@__DIR__, "loaded_depot")
+const FORCE_CLEAN_UNITS = false
+
+function test_depot_setup(; loaded_depot=false)
+    if FORCE_CLEAN_UNITS
+        return mktempdir()
+    end
+
+    Pkg.UPDATED_REGISTRY_THIS_SESSION[] = loaded_depot
+
+    test_depot = mktempdir()
+    if isdir(LOADED_DEPOT)
+        if loaded_depot
+            cp(LOADED_DEPOT, test_depot; force=true)
+        else
+            cp(joinpath(LOADED_DEPOT, "registries"), joinpath(test_depot, "registries"))#TODO should remove in final version?
+        end
+    end
+    return test_depot
+end
+
+function isolate(fn::Function; kwargs...)
```

Comment:
We should still probably have the`registry` URL hack in here that we have with `temp_pkg_dir` for tests where we do need a new `DEPOT` but don't want to close the registry from the internet anyway.

Reply:
Oh, I see the copy for registries now. Maybe that is faster.

Reply:
I think we will probably need the URL hack in some tests so that we can test that registries are being cached correctly. From what I can see, we are not doing that yet